lib/src/main/java/io/ably/lib/realtime/Presence.java

```diff
@@ -777,7 +835,7 @@ synchronized boolean hasNewerItem(String key, PresenceMessage item) {
 		 * @return
 		 */
 		synchronized Collection<PresenceMessage> values() {
-			try { return values(false); } catch (InterruptedException e) { return null; }
+			try { return values(false); } catch (InterruptedException|AblyException e) { return null; }
```

Comment:
From Java 7 only "try with resource" is not supported on all the targets, everything else is fine. Also I don't think it generates "special" byte code